framework/ldtest/test_scope.go

```diff
@@ -81,8 +81,8 @@ func (t *T) run(action func(*T)) {
 		if t.failed {
 			t.env.results.Failures = append(t.env.results.Failures, result)
 		}
-		for _, cleanupFn := range t.cleanups {
-			cleanupFn()
+		for i := len(t.cleanups) - 1; i >= 0; i-- {
+			t.cleanups[i]()
```

Comment:
This fixes a problem with test cleanup that I noticed in the logs. When a test calls methods like `NewSDKClient`, `NewSDKDataSource`, etc., we use the `t.Defer()` mechanism to ensure that those components are torn down on leaving the test scope. This should behave like Go's `defer` in that the most recently created things are torn down first, but it was doing the opposite, which could cause spurious errors.